src/App.tsx

```diff
@@ -1,60 +1,65 @@
-import classNames from 'classnames';
-
+import React, { useState, useEffect } from 'react';
+import './App.scss';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
-
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
-import { UserSelector } from './components/UserSelector';
-import { Loader } from './components/Loader';
+import { getAllUsers } from './api/api';
+import { User } from './types';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+const App: React.FC = () => {
+  const [allUsers, setAllUsers] = useState<User[]>([]);
+  const [currentUser, setCurrentUser] = useState('0');
+  const [selectedPostId, setSelectedPostId] = useState<number>();
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    getAllUsers()
+      .then(response => setAllUsers(response));
+  }, []);
```

Comment:
This is not a good approach, since it does not have any error treatments nor any loading state.

What if getAllUsers returns an error? We should be able to handle it and treat it somehow, reducing the impact of the error on the user experience.

What if getAllUsers's promise take more than 10 seconds to respond? we should be able to show its loading state to the user.

Another issue in this snippet is: Reusability!!!!

What if we want to use this method in another 10 screens? What we will do? copy it and paste it in those 10 files? NO!

We should be able to call this method from any component and use it in any screen, letting it be as reusable as possible (following the DRY principle), and, simultaneously, having its alternatives states.

We also have another issue here, and he's in the useEffect hook usage, it's purpose is to handling the side effects of the component, but we are using it gather data from the server, and we don't want to take risks and be open to issues like:

- Unintentional Requests on Re-render: If getAllUsers is defined or changes within the component, it will cause useEffect to rerun whenever getAllUsers changes, possibly leading to duplicate requests.
- Memory Leaks: There is no cleanup for when the component unmounts and the request is still ongoing, which could lead to setting state on an unmounted component if the request completes after the component is unmounted.


We can solve this by many ways, my suggestion is using the useQuery hook from [React Query](https://tanstack.com/query/latest/docs/framework/react/overview), which provides a simple and powerful solution for managing asynchronous data in React applications.